src/Selenium2Library/locators/elementfinder.py

```diff
@@ -23,19 +42,33 @@ def __init__(self):
             'scLocator': self._find_by_sc_locator,
             'default': self._find_by_default
         }
-        self._strategies = NormalizedDict(initial=strategies, caseless=True, spaceless=True)
+        self._strategies = NormalizedDict(initial=strategies, caseless=True,
+                                          spaceless=True)
         self._default_strategies = list(strategies.keys())
+        self._key_attrs = {
+            None: ['@id', '@name'],
+            'a': [
+                '@id', '@name', '@href',
+                'normalize-space(descendant-or-self::text())'
+            ],
+            'img': ['@id', '@name', '@src', '@alt'],
+            'input': ['@id', '@name', '@value', '@src'],
+            'button': [
+                '@id', '@name', '@value',
+                'normalize-space(descendant-or-self::text())'
+            ]
+        }
 
     def find(self, browser, locator, tag=None):
         assert browser is not None
         assert locator is not None and len(locator) > 0
 
-        (prefix, criteria) = self._parse_locator(locator)
-        prefix = 'default' if prefix is None else prefix
+        prefix, criteria = ParserLocator.parse_locator(locator)
         strategy = self._strategies.get(prefix)
         if strategy is None:
-            raise ValueError("Element locator with prefix '" + prefix + "' is not supported")
-        (tag, constraints) = self._get_tag_and_constraints(tag)
+            raise ValueError('Element locator with prefix \'{}\' '
+                             'is not supported'.format(prefix))
```

Comment:
- Why not double quotes around the string to avoid escaping the internal single quotes?

- As a very minor style thing, instead of using `self._strategies.get()` and validating that result isn't `None`, I'd use

   ```
    try:
        strategy = self._strategies[prefix]
    except KeyError:
        raise ValueError(...)
   ```

   or

   ```
    if prefix not in self._strategies:
        raise ValueError(...)
    strategy = self._strategies[prefix]
    ```